core/net/src/hybrid/cli.rs

```diff
@@ -85,3 +97,74 @@ fn to_status_metrics(metrics: &mut ChannelMetrics) -> model::StatusMetrics {
         rx_current: metrics.rx.short.average(time),
     }
 }
+
+pub async fn cli_ping(_msg: model::GsbPing) -> Result<Vec<GsbPingResponse>, StatusError> {
+    let client = {
+        CLIENT.with(|c| c.borrow().clone()).ok_or_else(|| {
+            model::StatusError::RuntimeException("client not initialized".to_string())
+        })?
+    };
+
+    // This will update sessions ping. We don't display them in this view
+    // but I think it is good place to enforce this.
+    client.ping_sessions().await;
+
+    let nodes = client.connected_nodes().await;
+    let our_node_id = client.node_id();
+    let ping_timeout = Duration::from_secs(10);
+
+    log::debug!("Ping: Num connected nodes: {}", nodes.len());
+
+    let results = join_all(
+        nodes
+            .iter()
+            .map(|(id, alias)| async move {
+                let tcp_before = Instant::now();
+
+                ya_net::from(our_node_id)
+                    .to(*id)
+                    .service(ya_net::DIAGNOSTIC)
+                    .send(GsbRemotePing {})
+                    .timeout(Some(ping_timeout))
+                    .await???;
+
+                let tcp_ping = tcp_before.elapsed();
```

Comment:
Can we assume that for each new session there exists an established vtcp connection? It's not a problem currently, but might be one if we introduce e.g. a `connect` command

Reply:
Even if we can't, I wouldn't like to handle this. What would we achieve? That someone avoids confusion, that ping takes longer due to TCP handshake?

Reply:
We would always print the actual response time. Handling it might be just as easy as sending the tcp ping message twice or making sure that a TCP connection is always established for each application session

Reply:
I would choose the second solution. But still at this point we don't have this problem